### PR TITLE
Update the codemeta file for the LTS release

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**
In the context of preparation of our first LTS release, the ``codemeta.json`` file should be updated accordingly to our new authorship policy (#3970).

This pull request contains an update of the metadata associated to the projet and an update of the author list.

**Dear reviewer**
The review should be made in two steps:
1. review of the general metadata
2. once the PR #4077 is accepted and merged, we will update the metadata associated to the authors

### CODE REVIEW

File: codemeta.json [hunk 1]
```diff
@@ -1,616 +1,809 @@
...
+    "runtimePlatform": [
+        "C",
+        "Cython",
+        "Python 3",
+        "Python 3.8",
+        "Python 3.9",
+        "Python 3.10",
+        "Python Implementation CPython"
+    ],
+    "softwareRequirements": [
```
Comment (registerrier) [inline codemeta.json:75]: The software requirements listed here correspond to the dev environment. Many are not needed.
Also should we put minimum requirements as per setup.cfg or the maximal list that will go in the `environment-v1.0.yaml`?

**Review by registerrier — COMMENTED** [review]
Thanks @bkhelifi .

The dependencies listed here are for the dev environment. What should be provide here? I suppose only the minimal required dependencies, no? So this should be synced with what in the setup.cfg. Right?

File: codemeta.json (hunk 1 above)
Comment (adonath) [reply codemeta.json:75]: I think this entry should be auto generated from the `setup.cfg` and just list the minimal requirements. The information in the `setup.cfg` is much more detailed. We shouldn't even try to duplicate it here min any way...

**Comment by registerrier** [comment]
Superseded by #4202

**Pull request closed without merging**